test/end2end.py

```diff
@@ -508,7 +508,7 @@ def test_openmpi4(self):
         self.assertTrue(regex.search(out), "Pattern '%s' should be found in: %s" % (regex.pattern, out))
 
         # mapping/binding to core is done by default
-        regex = re.compile(r"^fake mpirun called with args:.*--map-by core --bind-to core")
+        regex = re.compile(r"^fake mpirun called with args:.*--map-by ppr:2:node:PE=1:SPAN:NOOVERSUBSCRIBE")
```

Comment:
Why is this `ppr:2`? Will this test only pass on a dual-core system?

Reply:
Can we also add a test case where oversubscription is being allowed?

Reply:
Still not clear to me why this is `ppr:2` here, isn't this basically hardcoding to Jenkins which has 2 cores?